### PR TITLE
Add app.yml workflow

### PR DESCRIPTION
**Pull request by illume**
For testing the linux app build to start with.

I looked into seeing if there's a "--fail-with-warnings" that the electron builder gives us, which I hoped to use to catch issues like the icons missing one... but it doesn't exist.

I want to add a very simple test to that PR to see if the executable runs and then exits. A --run-and-exit type command, which runs and then exits. So we can confirm executable is working.

### CODE REVIEW

**Comment by joaquimrocha** [comment]
Can we do that without a command? Just maybe looking for the PID and killing it?

**Comment by illume** [comment]
Yes, maybe that could work too. I did a quick search-foo for and electron and signal handlers, and couldn't find much but I'm not clear if they would be a mess with electron/chromium. A command line argument(and/or environment variable) will give the possibility for the JavaScript code to run, and for the app to close gracefully at a specific point. Then we can check the return code for success.

**Comment by joaquimrocha** [comment]
I have pushed a few changes to fix the linux build + add the other 2 platforms.

File: app/mac/scripts/notarize.js [hunk 1]
```diff
@@ -2,8 +2,8 @@ require('dotenv').config();
 const { notarize } = require('electron-notarize');
 
 exports.default = async function notarizing(context) {
-  const { electronPlatformName, appOutDir } = context;  
-  if (electronPlatformName !== 'darwin') {
+  const { electronPlatformName, appOutDir } = context;
+  if (electronPlatformName !== 'darwin' || process.env.CI === 'true') {
```
Comment (illume) [inline app/mac/scripts/notarize.js:6]: Not related to this PR... but this reminds me: If you don't have the notarization files the app-mac build currently breaks. I'll make an issue/PR.

File: app/package.json [hunk 2]
```diff
@@ -65,9 +58,7 @@
     },
     "win": {
       "target": [
-        "zip",
-        "msi",
```
Comment (illume) [inline app/package.json:69]: Another semi related comment, but msi is deprecated from around 2018/2019 I think.

**Comment by illume** [comment]
Just realized this is my own PR, so I can't review it...
so here's a 👍 from me.

File: app/mac/scripts/notarize.js (hunk 1 above)
Comment (joaquimrocha) [reply app/mac/scripts/notarize.js:6]: You mean the notarization credentials, right? This line is runing them off for the CI. An alternative is that disable it if the APPLE* stuff used below is not set up.

**Pull request merged**